iothub/device/src/Transport/Amqp/AmqpAuthenticationRefresher.cs

```diff
@@ -64,9 +65,28 @@ public async Task InitLoopAsync(TimeSpan timeout)
                     timeout)
                 .ConfigureAwait(false);
 
+            // This cancellation token source is disposed when the authentication refresher is disposed
+            // or if this code block is executed more than once per instance of AmqpAuthenticationRefresher (not expected).
```

Comment:
If it isn't expected you can instead opt for a `Debug.Assert(_refresherCancellationTokenSource == null);` instead. The idea being that we catch it in testing.